src/app/signin/page.tsx

```diff
@@ -0,0 +1,280 @@
+		setConfirmPasswordError('');
+	};
+
+	const onUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setUsername(event.target.value);
+		setUsernameError('');
+	};
+
+	const onFirstNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setFirstName(event.target.value);
+	};
+
+	const onLastNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setLastName(event.target.value);
+	};
+
+	const validateForm = () => {
+		let isValid = true;
+
+		if (!username) {
+			setUsernameError('Please enter a username.');
+			isValid = false;
+		}
+
+		if (!email) {
+			setEmailError('Please enter an email address.');
+			isValid = false;
+		}
+
+		if (!password) {
+			setPasswordError('Please enter a password.');
+			isValid = false;
+		}
+
+		if (password !== confirmPassword) {
+			setConfirmPasswordError('Passwords do not match.');
+			isValid = false;
+		}
+
+		return isValid;
+	};
+
+	const onSubmit = (event: React.FormEvent) => {
+		event.preventDefault();
+		if (validateForm()) {
+			register();
+		}
+	};
+
+	const toggleLoader = (state: boolean) => {
+		const button: HTMLElement | null = document.getElementById("signin-button");
+
+		if (button)
+			button.classList.toggle("loading", state);
+	};
+
+	const [register, { loading: registerLoading, error: registerError }] = useMutation(REGISTER_MUTATION, {
+		variables: {
+			email: email,
+			password: confirmPassword,
+			username: username,
+			firstName: firstName,
+			lastName: lastName,
+		},
+		onCompleted: () => {
+			toggleLoader(false);
+			Notify.success(`Thanks for register. Please, confirm your email address!`);
+			router.push('/login')
+		},
+		onError: (error) => {
+			toggleLoader(false);
+			Notify.failure(`${error.message}!`);
+			setErrorMsg(error.message);
+		}
+	});
+
+	if (registerLoading) {
+		toggleLoader(true);
+	}
+	if (registerError) {
+		toggleLoader(false);
+		errorMsg;
+	}
+
+	// Check mark and X mark for password validation
+	const checkMark = <span className="text-green-600">&#10004;</span>;
+	const xMark = <span className="text-red-600">&#x2717;</span>;
+
+	return (
+		<div className="flex flex-col h-screen">
+			<div className="grid grid-cols-2 p-5">
+				<div className="flex">
+					<p className="text-5xl font-montez">Cursif</p>
+				</div>
+				<div className="flex items-center justify-end">
+					<a href="/login" className="button"><span className="label">Log In</span></a>
+				</div>
+			</div>
+
+			<div className="flex-1 p-5">
+				<div className="flex justify-center h-full">
+					<form className="w-[400px]" onSubmit={onSubmit}>
+						<div className="text-center mt-20">
+							<h1 className="text-5xl"><b>SIGN</b> IN</h1>
+						</div>
+
+						<div className="my-8">
+							<div className="flex row my-5">
+								<input
+									className="input w-full mr-3"
+									type="text"
+									placeholder="First Name"
+									value={firstName}
+									onChange={onFirstNameChange}
+									required={false}
```

Comment:
Why this field and the next one aren't required anymore? Is it just an oversight? Anyway, those should be required.

Reply:
The first and last name field requirements are set to false because it isn’t a requirement but is optional for registration (that’s how we configured the [backend](https://github.com/Code-Society-Lab/cursif-backend/blob/06ef89f50d28910a1890e8df0894118e35ae0178/lib/cursif_web/schemas/account.ex#L28C7-L28C7)).